authentication-service/src/main/kotlin/com/saveourtool/save/authservice/config/WebSecurityConfig.kt

```diff
@@ -0,0 +1,136 @@
+import org.springframework.security.config.annotation.method.configuration.EnableReactiveMethodSecurity
+import org.springframework.security.config.annotation.web.reactive.EnableWebFluxSecurity
+import org.springframework.security.config.web.server.SecurityWebFiltersOrder
+import org.springframework.security.config.web.server.ServerHttpSecurity
+import org.springframework.security.crypto.factory.PasswordEncoderFactories
+import org.springframework.security.crypto.password.PasswordEncoder
+import org.springframework.security.web.server.SecurityWebFilterChain
+import org.springframework.security.web.server.authentication.AuthenticationWebFilter
+import org.springframework.security.web.server.authentication.HttpStatusServerEntryPoint
+
+import javax.annotation.PostConstruct
+
+@EnableWebFluxSecurity
+@EnableReactiveMethodSecurity
+@Profile("secure")
+@Suppress("MISSING_KDOC_TOP_LEVEL", "MISSING_KDOC_CLASS_ELEMENTS", "MISSING_KDOC_ON_FUNCTION")
+class WebSecurityConfig(
+    private val authenticationManager: ConvertingAuthenticationManager,
+    @Autowired private var defaultMethodSecurityExpressionHandler: DefaultMethodSecurityExpressionHandler
+) {
+    @Bean
+    fun securityWebFilterChain(
+        http: ServerHttpSecurity
+    ): SecurityWebFilterChain = http.run {
+        authorizeExchange()
+            .pathMatchers(*publicEndpoints.toTypedArray())
+            .permitAll()
+            // resources for frontend
+            .pathMatchers("/*.html", "/*.js*", "/*.css", "/img/**", "/*.ico", "/*.png", "/particles.json")
+            .permitAll()
+    }
+        .and()
+        .run {
+            authorizeExchange()
+                .pathMatchers("/**")
+                .authenticated()
+        }
+        .and()
+        .run {
+            // FixMe: Properly support CSRF protection https://github.com/saveourtool/save-cloud/issues/34
+            csrf().disable()
+        }
+        .addFilterBefore(
+            AuthenticationWebFilter(authenticationManager).apply {
+                setServerAuthenticationConverter(CustomAuthenticationBasicConverter())
+            },
+            SecurityWebFiltersOrder.HTTP_BASIC,
+        )
+        .exceptionHandling {
+            it.authenticationEntryPoint(
+                HttpStatusServerEntryPoint(HttpStatus.UNAUTHORIZED)
+            )
+        }
+        .logout()
+        .disable()
+        .formLogin()
+        .disable()
+        .build()
+
+    @PostConstruct
+    fun postConstruct() {
+        defaultMethodSecurityExpressionHandler.setRoleHierarchy(roleHierarchy())
+    }
+
+    companion object {
+        /**
+         * These endpoints will have `permitAll` enabled on them. We can't selectively put `@PreAuthorize("permitAll")` in the code,
+         * because it won't allow us to configure authenticated access to all other endpoints by default.
+         * Or we can use custom AccessDecisionManager later.
+         */
+        internal val publicEndpoints = listOf(
```

Comment:
Maybe we should have separate lists of public endpoints for different services? It doesn't make much sense in a common configuration IMO.